hedy.py

```diff
@@ -174,6 +174,17 @@ def needs_colon(rule):
     return f'{rule[0:pos]} _COLON {rule[pos:]}'
 
 
+def translate_index_error(code, liste_name):
+    exception_text = gettext('catch_index_exception').replace('{list_name}', style_command(liste_name))
+    exception_text = exception_text.replace("'", "\\'")
+    return textwrap.dedent(f"""\
+        try:
+          {code}
+        except IndexError:
+          raise Exception('{exception_text}')
```

Comment:
@Felienne Not strictly part of this change, so not sure if we want to include it, but we can make this `raise Exception({repr(exception_text)})` to properly escape all string oddities. Then we no longer need the `replace` method 
for the single quote escaping above. Let me know if you want to add this here.

Reply:
Lovely idea, that would allow us to catch other issues too. I am ok with you pushing this on this branch but it is also ok of course if @etienneCharignon does that!

Reply:
very well ! That's what I wanted to do initaily but did not know how to do it.